docs/community/rust-coding-guidelines/rust-secure-coding-guidelines.md

```diff
@@ -0,0 +1,253 @@
+
+### Rustfix
+
+Apply `rustfix` for compiler warnings and edition transitions, but verify the automatic fixes to ensure that the recommendations match the purpose of the code [[rustfix]](https://anssi-fr.github.io/rust-guide/02_devenv.html#rustfix).
+
+### Documentation
+
+Document safety invariants and security considerations in code, especially for public and `unsafe` functions.
+
+````rust
+  foo(
+      // SAFETY:
+      // This is a valid safety comment
+      unsafe { *x }
+  )
+  ```
+
+  ```rust
+  use std::ptr::NonNull;
+  let a = &mut 42;
+
+  // SAFETY: references are guaranteed to be non-null.
+  let ptr = unsafe { NonNull::new_unchecked(a) };
+````
+
+## Libraries
+
+### Crate Quality and Security
+
+Assess and monitor the quality and maintenance of used crates, employing tools like `cargo-outdated` and `cargo-audit` for version management and vulnerability checking [[audit lib]](https://anssi-fr.github.io/rust-guide/03_libraries.html#libraries).
+
+- Aptos utilizes **[Dependabot](https://github.com/dependabot)** to continuously monitor libraries. Our policy requires mandatory updates for critical and high-vulnerabilities, or upon impact evaluation given the context for medium and lower.
+- When vetting a new library, it's advisable to utilize deps.dev for evaluation. This site provides an OpenSSF scorecard containing essential information. As a guideline, libraries with an OpenSSF score of 7 or higher are typically safe to import. However, those scoring **below 7** must undergo discussion prior to importation.
+
+### Minimize Use of Feature Flags
+
+As a general practice, avoid using feature flags in your crates unless absolutely necessary. Feature flags can introduce complexity and unexpected behaviours, making the codebase harder to audit for security vulnerabilities.
+
+### Understanding Feature Unification
+
+Be aware of Cargo's feature unification process. When multiple dependencies require the same crate with different feature flags, Cargo unifies these into a single configuration. This unification can inadvertently enable features that might not be desirable or secure for the project [[Rustbook: features unification]](https://doc.rust-lang.org/cargo/reference/features.html#feature-unification) [[Rustbook: feature resolver]](https://doc.rust-lang.org/cargo/reference/features.html#feature-resolver-version-2).
+
+## Language Generalities
+
+### Unsafe Code
+
+Never use `unsafe` blocks unless as a last resort. Justify their use in a comment, detailing how the code is effectively safe to deploy [[unsafe code]](https://anssi-fr.github.io/rust-guide/04_language.html#unsafe-code).
+
+### Integer Overflows
+
+Refer to [coding-guidelines](./rust-coding-guidelines.md#integer-arithmetic).
```

Comment:
this moves the attention to another document. can we replicate/reframe the content here with a security focus?

Reply:
The idea was to don't duplicate material between coding guidelines and secure coding.